lib/resty/healthcheck.lua

```diff
@@ -1011,7 +1023,7 @@ end
 -- Log a message specific to this checker
 -- @param level standard ngx log level constant
 function checker:log(level, ...)
-  ngx_log(level, worker_color(self.LOG_PREFIX), ...)
+  return ngx_log(level, worker_color(self.LOG_PREFIX), ...)
```

Comment:
this does not seem necessary, as we don't care about the return of ngx.log (is there any? the docs don't say so)

Reply:
We use lua's tail calling feature here so that we can see the caller's line number in the error log.

please take a look at this example, pay more attention about the output of error log:

```shell
$ resty -e 'local function log1(...)
                ngx.log(ngx.ERR, ...)
            end

            local function log2(...)
                return ngx.log(ngx.ERR, ...)
            end

            log1("first")
            log2("second")'
2019/07/12 21:04:01 [error] 40647#40647: *2 [lua] (command line -e):2: log1(): first, context: ngx.timer
2019/07/12 21:04:01 [error] 40647#40647: *2 [lua] (command line -e):10: second, context: ngx.timer
```

I think we more love the implementation of `log2`, which can record the original source code line number.

Reply:
@membphis that is a neat trick to get the proper line logged. It has been very annoying to only get the lines from the log function. Definitely one to remember and reuse elsewhere. Thx.

Reply:
@membphis @Tieske while Lua does have a tail call feature, losing the line number info when performing a tail call is a LuaJIT-specific (mis)behavior. Here's an equivalent non-resty specific script:

```
local function x()
   return debug.getinfo(1, "l")
end

local function y()
   local x = debug.getinfo(1, "l")
   return x
end

print(x().currentline)
print(y().currentline)
```

With Lua 5.3 this prints "2 6", as it should; with `resty` this prints "10 6".

It may be a "neat trick" that you managed to get a useful behavior out of the fact that LuaJIT loses debugging information, but I would be wary of relying on such undefined behavior. A future version of the compiler may fix this bug and then line number info will change in surprising ways.

Reply:
To clarify, I'm not saying this change should be removed from this PR — just that I'm not a fan of it :)

Reply:
It seems that we can keep this improvement. After all, the project works in the openresty environment, we can only use luajit.